src/ca.c

```diff
@@ -2256,15 +2035,15 @@ static int ci_ccmgr_cc_sac_send(struct ci_session *session, const uint8_t *tag,
     sac_crypt(&data[8], &data[8], pos - 8, cc_data->sek, AES_ENCRYPT);
 
     //        _hexdump("ENCRYPTED    ",data, pos);
-    ci_session_sendAPDU(session, tag, data, pos);
+    ca_write_apdu(d, tag, data, pos);
 
     return 1;
 }
 
-static int ci_ccmgr_cc_sac_data_req(ca_device_t *d, struct ci_session *session,
-                                    const uint8_t *data, unsigned int len) {
-    struct cc_ctrl_data *cc_data = session->private_data;
-    const uint8_t data_cnf_tag[3] = {0x9f, 0x90, 0x08};
+static int ci_ccmgr_cc_sac_data_req(ca_device_t *d, const uint8_t *data,
+                                    unsigned int len) {
+    struct cc_ctrl_data *cc_data = &d->private_data;
+    uint32_t data_cnf_tag = 0x9f9008;
```

Comment:
There are DEFINEs for most tags, IMO we should use defines for these as well instead of magic numbers